src/envoy/admin/crud/site_reading.py

```diff
@@ -0,0 +1,99 @@
+from datetime import datetime
+from typing import Sequence
+
+from sqlalchemy import func, select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+
+from envoy.server.model.site_reading import SiteReading, SiteReadingType, Site
+from envoy_schema.server.schema.sep2.types import DataQualifierType, KindType
+
+
+async def select_agg_id_for_site(
+    session: AsyncSession,
+    site_id: int,
+) -> Sequence[int]:
+    """Function to get aggregator id for a site, usually to help hit an index"""
+
+    stmt = select(Site.aggregator_id).where(Site.site_id == site_id)
+
+    resp = await session.execute(stmt)
+
+    return resp.scalars().all()
+
+
+async def count_site_readings_for_site_and_time(
+    session: AsyncSession,
+    site_type_ids: Sequence[int],
+    start_time: datetime,
+    end_time: datetime,
+) -> int:
+    """Count total site readings for a sequence of site_type_ids within a time range."""
+
+    # Return 0 immediately if no site_type_ids provided
+    if not site_type_ids:
+        return 0
+
+    stmt = (
+        select(func.count(SiteReading.site_reading_id))
+        .where(SiteReading.site_reading_type_id.in_(site_type_ids))
+        .where(SiteReading.time_period_start >= start_time)
+        .where(SiteReading.time_period_start < end_time)
+    )
+
+    resp = await session.execute(stmt)
+    return resp.scalar() or 0
+
+
+async def select_csip_aus_site_type_ids(
+    session: AsyncSession,
+    aggregator_ids: Sequence[int],
+    site_id: int,
+    uom: int,
```

Comment:
It would be nicer if uom aligned with the int enum UomType defined on the SiteReadingType

from envoy_schema.server.schema.sep2.types import UomType